src/plugins/controls/common/types.ts

```diff
@@ -27,3 +27,8 @@ export type ControlInput = EmbeddableInput & {
   controlStyle?: ControlStyle;
   ignoreParentSettings?: ParentIgnoreSettings;
 };
+
+export type DataControlInput = ControlInput & {
+  fieldName: string;
+  dataViewId: string;
+};
```

Comment:
Currently, **all** of the controls we have share the following properties: `fieldName`, `dataViewId`, and a new public function `resetSelections` (introduced as part of this PR) that will allow external actions to clear the selections of a given control embeddable.

While this is true **for now**, in the future we may have different types of controls - for example, those that set dashboard variables or ones that change configuration options - and these new controls may not need all three of these properties. Therefore, rather than simply making these three properties part of the generic `ControlInput`, I defined a **seperate** type for data-specific controls `DataControlInput` that all of our **current** controls should use instead of the generic `ControlInput`